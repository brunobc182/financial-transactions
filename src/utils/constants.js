const TRANSACTION_TYPE = {
  DEBIT: 'DEBIT',
  CREDIT: 'CREDIT',
};

export default TRANSACTION_TYPE;
