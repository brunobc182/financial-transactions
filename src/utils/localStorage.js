const setItem = (name, item) => localStorage.setItem(name, JSON.stringify(item));
const getItem = name => JSON.parse(localStorage.getItem(name));

export { setItem, getItem };
