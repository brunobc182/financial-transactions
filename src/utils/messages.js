const MESSAGES = {
  totalLabel: 'Total: ',
  partialTotalLabel: 'Partial total: ',
};

export default MESSAGES;
