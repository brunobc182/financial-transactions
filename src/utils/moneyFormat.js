const moneyFormat = (number, prefix = '$') => `${prefix} ${number
  .toFixed(2)
  .replace(/(\d)(?=(\d{3})+\.)/g, '$1.')
  .replace(/\.(\d{2})$/, ',$1')}`;

export default moneyFormat;
