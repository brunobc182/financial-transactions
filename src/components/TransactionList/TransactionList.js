import React from 'react';
import styled from 'styled-components';
import { node, array, oneOfType } from 'prop-types';
import colors from '../../theme/colors';
import { moneyFormat } from '../../utils';

const TransactionList = ({ data }) => (
  <ListWrapper>
    {data
      && data.map(({ type, value }, index) => (
        <ListItem key={`id${data.length - index}`} type={type}>
          <ListItemContent>{moneyFormat(value, '$')}</ListItemContent>
        </ListItem>
      ))}
  </ListWrapper>
);
const ListWrapper = styled.ol`
  width: 100%;
  padding: 0;
`;

const ListItem = styled.li`
  text-align: center;
  list-style-position: inside;
  width: 100%;
  height: 48px;
  font-size: 22px;
  color: ${colors.white};
  background-color: ${({ type }) => (type === 'DEBIT' ? colors.radicalRed : colors.shamrock)};
  box-shadow: 0 6px 10px 0 rgba(0, 0, 0, 0.1);
  margin: 5px 0;
  border-radius: 4px;
`;
const ListItemContent = styled.p`
  display: inline-block;
  margin: 10px;
  padding: 0;
`;

TransactionList.propTypes = {
  data: oneOfType([node, array]).isRequired,
};
export default TransactionList;
