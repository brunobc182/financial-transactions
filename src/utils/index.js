import TRANSACTION_TYPE from './constants';
import moneyFormat from './moneyFormat';
import MESSAGES from './messages';
import { getItem, setItem } from './localStorage';

export {
  TRANSACTION_TYPE, moneyFormat, MESSAGES, getItem, setItem,
};
