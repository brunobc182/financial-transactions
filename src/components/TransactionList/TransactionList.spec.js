import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import sinon from 'sinon';
import { TRANSACTION_TYPE } from '../../utils';
import TransactionList from './TransactionList';

describe('TransactionList', () => {
  const { CREDIT, DEBIT } = TRANSACTION_TYPE;
  const data = [
    {
      value: 10,
      type: CREDIT,
      partialTotal: 10,
    },
  ];
  it('should render correctly', () => {
    const tree = renderer.create(<TransactionList data={data} onClick={Function} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with type DEBIT prop', () => {
    const newData = [
      {
        value: 10,
        type: DEBIT,
        partialTotal: 10,
      },
    ];
    const tree = renderer
      .create(<TransactionList data={newData} onClick={Function} isDebit />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onClick function', () => {
    const spy = sinon.spy();
    const tree = mount(<TransactionList data={data} onClick={spy} />);
    tree.find('button').simulate('click');
    expect(spy.called).toBe(true);
  });
});
