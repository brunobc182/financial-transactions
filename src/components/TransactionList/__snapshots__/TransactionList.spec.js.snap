// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionList should render correctly 1`] = `
<ol
  className="sc-bdVaJa eKiYv"
>
  <li
    className="sc-bwzfXH dwTtjc"
    id={0}
    type="DEBIT"
  >
    <p
      className="sc-htpNat hiCggQ"
    >
      $ 10,00
    </p>
    <button
      className="sc-bxivhb eJBkbL"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </li>
</ol>
`;
